### ISSUE TITLE
Image instead of black bars - VisualServer

### ISSUE THREAD

**Issue by ParradoxSVK** [godotengine/godot#27440]
**Godot version:**
3.1 stable

**OS/device including version:**
Windows 10 

**Issue description:**
I am having problem setting texture to be used instead of black bars when application window is scaled. It always draws one color of texture(instead of the texture itself) like if it was zoomed in or something. Is this normal behaviour or a bug? I tried textures with defferent resolutions and also images instead of textures but nothing seems to work. 

Example - godot icon is used as texture for black bars
![godot_window](https://user-images.githubusercontent.com/48955019/55005970-3fe74a80-4fdd-11e9-8fbc-f66fdf5d294e.png)

Normally i would expect that texture that is passed to visualServer will be scaled to the size corresponding bar instead of this.

**Steps to reproduce:**
```
var txt = preload("res://icon.png")

func _ready():
	VisualServer.black_bars_set_images( txt .get_rid(),  txt .get_rid(),  txt .get_rid(),  txt .get_rid())
	VisualServer.black_bars_set_margins(50,50,50,50)
```

**Comment by azagaya** [comment]
Hi! I made a fix in [rasterizer_canvas_gles3.cpp](https://github.com/godotengine/godot/blob/3.2/drivers/gles3/rasterizer_canvas_gles3.cpp) in 3.2 branch. The result is this:
![border](https://user-images.githubusercontent.com/46932830/77851186-ff04b180-71ad-11ea-934b-dca7e74ee9aa.gif)
However, idk if this is the desired result. It just preserves the size and aspect of the images.
Also, idk how to test in vulkan branch.

**Comment by azagaya** [comment]
Should i make a PR for this for 3.2?

**Comment by ParradoxSVK** [comment]
Well i would say go for it. It looks pretty good to me.

**Comment by azagaya** [comment]
Perhaps i could also add the option to stretch the image to the margins instead of preserving the size.. would it be useful?

**Comment by ParradoxSVK** [comment]
Well it might have it`s use cases for sure. I don`t see a problem with adding that option too.

**Comment by azagaya** [comment]
I created the PR... lets see if it get merged, as it would only be useful for 3.2 (4.0 doesn't have gles3 anymore).

**Comment by akien-mga** [comment]
@azagaya Do you have a project that can be used to test the fix? I'm trying with a couple lines as in #37475, but even with the fix I still get some pretty bad artifacts (even though it's much better with this fix that without).

**Comment by akien-mga** [comment]
Fixed by #37475.

**Comment by azagaya** [comment]
> @azagaya Do you have a project that can be used to test the fix? I'm trying with a couple lines as in the OP, but even with the fix I still get some pretty bad artifacts (even though it's much better with this fix that without).

I'll make one after work today. I recently changed my notebook and lost the example project in the way.
